dynamodb/src/main/scala/zio/dynamodb/Codec.scala

```diff
@@ -572,11 +632,82 @@ private[dynamodb] object Codec {
       case av                        => Left(s"unable to decode $av as a list")
     }
 
+    private def setDecoder[A](s: Schema[A]): Decoder[Set[A]] = {
+      def nativeStringSetDecoder[A]: Decoder[Set[A]] = {
+        case AttributeValue.StringSet(stringSet) =>
+          Right(stringSet.asInstanceOf[Set[A]])
+        case av                                  =>
+          Left(s"Error: expected a set but found '$av'")
```

Comment:
Do we want the error here to say `... expected a string set but ...`?

Reply:
:+1: